controllers/helpers.go

```diff
@@ -246,6 +246,96 @@ func CreateSystemDiskSpec(imageUUID string, systemDiskSize int64) (*prismclientv
 	return systemDisk, nil
 }
 
+// CreateDataDiskList creates a list of data disks with the given data disk specs
+func CreateDataDiskList(ctx context.Context, client *prismclientv3.Client, dataDiskSpecs []infrav1.NutanixMachineVMDisk) ([]*prismclientv3.VMDisk, error) {
+	dataDisks := make([]*prismclientv3.VMDisk, 0)
+
+	latestDeviceIndexByAdapterType := make(map[string]int64)
+	getDeviceIndex := func(adapterType string) int64 {
+		if latestDeviceIndex, ok := latestDeviceIndexByAdapterType[adapterType]; ok {
+			latestDeviceIndexByAdapterType[adapterType] = latestDeviceIndex + 1
+			return latestDeviceIndex
+		}
+
+		if adapterType == string(infrav1.NutanixMachineDiskAdapterTypeSCSI) || adapterType == string(infrav1.NutanixMachineDiskAdapterTypeIDE) {
+			latestDeviceIndexByAdapterType[adapterType] = 1
+			return 1
+		} else {
+			latestDeviceIndexByAdapterType[adapterType] = 0
+			return 0
+		}
+	}
+
+	for _, dataDiskSpec := range dataDiskSpecs {
+		dataDisk := &prismclientv3.VMDisk{
+			DiskSizeMib: utils.Int64Ptr(GetMibValueOfQuantity(dataDiskSpec.DiskSize)),
+		}
+
+		// If data source is provided, get the image UUID
+		if dataDiskSpec.DataSource != nil {
+			imageUUID, err := GetImageUUID(ctx, client, dataDiskSpec.DataSource.Name, dataDiskSpec.DataSource.UUID)
```

Comment:
This conflicts with #489. Since that's been open for over a month, and approved by one reviewer, I would ask that we merge it first, and rebase. I can help resolve the merge conflict.